File: HorseRacing.Contracts/Models/Game/Requests/PlaceBetRequest.cs
using HorseRacing.Contracts.Base.Requests;
using HorseRacing.Contracts.Models.Game.Dtos;

namespace HorseRacing.Contracts.Models.Game.Requests
{
    public class PlaceBetRequest : BaseRequest<PlaceBetRequestDto>
    {
        public PlaceBetRequest() : base() { }
        public PlaceBetRequest(PlaceBetRequestDto data) : base(data) { }
    }
}

File: HorseRacing.Web/src/interfaces/api/contracts/model/game/requests/place-bet/place-bet-request.ts
import type { BaseRequest } from "~/interfaces/api/contracts/base/requests/base-request";
import type { PlaceBetRequestDto } from "./place-bet-request-dto";

export class PlaceBetRequest implements BaseRequest<PlaceBetRequestDto> {
    Data: PlaceBetRequestDto;

    constructor(data?: PlaceBetRequestDto) {
        this.Data = data || { UserId: '', GameId: '', BetAmount: 0, BetSuit: 0 };
    }
}
